src/PgNamed.hs

```diff
@@ -264,6 +265,29 @@ executeNamed conn qNamed params =
     withNamedArgs qNamed params >>= \(q, actions) ->
         liftIO $ PG.execute conn q (toList actions)
 
+{- | Modifies the database with a given query and named parameters
+and returns the specified via @RETURNING@ keyword result.
+
+@
+'returningNamed' dbConnection [sql|
+    __UPDATE__ table
+    __SET__ foo = \'bar\'
+    __WHERE__ id = ?id
+    __RETURNING__ id
+|] [ "id" '=?' someId ]
+@
+-}
+
+returningNamed
+    :: (MonadIO m, WithNamedError m, PG.FromRow res)
+    => PG.Connection  -- ^ Database connection
+    -> PG.Query       -- ^ Query with named parameters inside
+    -> [NamedParam]   -- ^ The list of named parameters to be used in the query
+    -> m [res]        -- ^ Returning result
+returningNamed conn qNamed params =
+    withNamedArgs qNamed params >>= \(q, actions) ->
+        liftIO $ PG.returning conn q [toList actions]
```

Comment:
@vrom911 Yeah, this is a difficult problem. The same problem is for the `executeMany` function. That's why we didn't manage to implement it in the `mysql-not-so-simple` library. The problem is in the fact, as you correctly noticed, that `returning` and `executeMany` work with multiple rows, not a single one. So they allow to INSERT/UPDATE many values at once. And with plain `returning` we would write code like this:

```haskell
returning conn [sql|...|] [User "foo" 42, User "bar" 69]
```

But with named  version we need to write like this:

```haskell
```haskell
returning conn [sql|...|] [ ["name" =? "foo", age =? 42], ["name" =? "bar", age =? 69] ]
```

The only way to implement this that I managed to think of is to pass a function `a -> [NamedParam]` as an argument so the calls will look like this:

```haskell
returningNamed
    conn
    [sql|...|] 
    (\User{..} -> [ "name" =? userName, "age" =? userAge ])
    someUsers
```

But I'm not sure whether this is actually works.

Reply:
Another possible way could be:

```haskell
returningNamed
    conn
    [sql|...|]
    [ "name" =? [user1, user2]
    , "age" =? [age1, age2]
    ]
```

Slightly easier but introduces an illegal state where the lists are not of the same length.